xoinvader/animation.py

```diff
@@ -134,3 +151,57 @@ def update(self):
         if self._timer.get_elapsed() >= keyframe[0]:
             setattr(self._obj, self._attr, keyframe[1])
             self._current += 1
+def linear_equation(val1, val2, time1, time2, current_time):
+    """Linear equation to get interpolated value.
+
+    :param float val1: first keyframe value
+    :param float val2: second keyframe value
+    :param float time1: first keyframe local time
+    :param float time2: second keyframe local time
+    :param float current_time: current animation local time
+    """
+
+    return val1 + (val2 - val1) / (time2 - time1) * (current_time - time1)
+
+
+def same_type(values, types):
+    """Check if values are instances of same type.
+
+    :param collections.Iterable values: values to check type similarity
+    :param tuple|type types: type or tuple of types
+    """
+
+    return all(map(lambda it: isinstance(it, types), values))
+
+
+def interpolate(first, second, current_time):
+    """Interpolate value by two bounding keyframes.
+
+    :param collections.Iterable first: first bounding keyframe
+    :param collections.Iterable second: second bounding keyframe
+    :param float current_time: current animation local time
+
+    :raises AnimationBoundariesExceeded: when time interval is invalid
+    :raises InterpolationUnknownTypes: when interpolating invalid types
+    """
+
+    if not first[0] <= current_time <= second[0]:
+        raise AnimationBoundariesExceeded(first[0], current_time, second[0])
+
+    def frames_of(types):
```

Comment:
Why not use `*args` here? It could help to get rid of double parens when we pass type tuple and not a single type.